### PR TITLE
Feature/tvist1 164 hearing start

### PR DESCRIPTION
**Pull request by jekuaitk**
* Added `Hearing` and `HearingPost` entities
* Added logic for starting hearing and creating entity posts

https://jira.itkdev.dk/browse/TVIST1-164

### CODE REVIEW

File: src/Entity/Hearing.php [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+    /**
+     * @ORM\Column(type="boolean", options={"default":"0"})
+     */
+    private $hasNewHearingPost = false;
+
+    /**
+     * @ORM\Column(type="boolean", options={"default":"0"})
+     */
+    private $hasBeenStarted = false;
```
Comment (rimi-itk) [inline src/Entity/Hearing.php:54]: I suggest using a `?DateTimeInterface` value here (and renaming to `startedAt`). It's always nice to (be able to) know when something was started in addition to know it being started.

File: src/Entity/Hearing.php [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+    /**
+     * @ORM\Column(type="date", nullable=true)
+     */
+    private $startDate;
+
+    /**
+     * @ORM\Column(type="boolean", options={"default":"0"})
+     */
+    private $hasFinished = false;
```
Comment (rimi-itk) [inline src/Entity/Hearing.php:64]: Consider using a `?DateTimeInterface` value here.

File: src/Entity/HearingPost.php [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+    /**
+     * @ORM\Column(type="text")
+     */
+    private $content;
+
+    /**
+     * @ORM\Column(type="boolean", options={"default":"0"})
+     */
+    private $hasBeenProcessedAndForwarded = false;
```
Comment (rimi-itk) [inline src/Entity/HearingPost.php:45]: Consider using a `?DateTimeInterface` value here.

File: src/Entity/MailTemplate.php [hunk 4]
```diff
@@ -42,7 +42,6 @@ class MailTemplate
      * @ORM\Column(type="string", length=255)
      *
      * @var string
-     * @Assert\NotBlank()
```
Comment (rimi-itk) [inline src/Entity/MailTemplate.php:45]: 👍

File: templates/case/hearing/document_upload.html.twig [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        </div>
+        <div>
+            <label >{% trans %}Chose a file{% endtrans %}</label>
+            {{ form_widget(document_form.filename, {'attr': {'type': 'file', 'class': 'form-control-file', 'aria-describedby': 'helpFileId', 'required': true}}) }}
+            <small id="helpFileId" class="form-text text-muted">{% trans %}Max file size: 10mb. File formats accepted: .pdf, .txt, .mp4, .jpeg, .png, .doc, .xls{% endtrans %}</small>
+        </div>
+        <button class="btn btn-success float-right">{{ button_label|default('Upload'|trans([], 'documents', 'da')) }}</button>
+        {{ form_rest(document_form) }}
+    </main>
+{% endblock %}
```
Comment (rimi-itk) [inline templates/case/hearing/document_upload.html.twig:34]: This should look more like https://github.com/itk-dev/naevnssekretariatet/blob/develop/templates/documents/create.html.twig.

**Review by rimi-itk — COMMENTED** [review]
Some suggestions added.

File: src/Controller/HearingController.php [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+     * @Route("/{id}/hearing", name="case_hearing_index")
+     */
+    public function hearing(CaseEntity $case, HearingPostRepository $hearingPostRepository, PartyHelper $partyHelper, Request $request): Response
+    {
+        $this->denyAccessUnlessGranted('edit', $case);
+
+        // Check whether there is at least one part for each side
+        $relevantParties = $partyHelper->getRelevantPartiesByCase($case);
+
+        $hasSufficientParties = sizeof($relevantParties['complainants']) > 0 && sizeof($relevantParties['counterparties']) > 0;
```
[the remark this answers is not in the document]
Comment (jekuaitk) [reply src/Controller/HearingController.php]: Fixed in aaf76161726183f5f8abb570034339783963f2d0.

File: src/Controller/HearingController.php [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+
+        if (null === $hearing) {
+            $hearing = new Hearing();
+            $case->setHearing($hearing);
+
+            $this->entityManager->persist($hearing);
+            $this->entityManager->flush();
+        }
+
+        $neitherPartyHasAnythingToAdd = true === $hearing->getComplainantHasNoMoreToAdd() && true === $hearing->getCounterpartHasNoMoreToAdd();
```
[the remark this answers is not in the document]
Comment (jekuaitk) [reply src/Controller/HearingController.php:53]: Nope, fixed in aaf76161726183f5f8abb570034339783963f2d0.

File: templates/case/hearing/document_upload.html.twig (hunk 5 above)
Comment (jekuaitk) [reply templates/case/hearing/document_upload.html.twig]: This is unused, removed in aaf76161726183f5f8abb570034339783963f2d0.

File: src/Entity/Hearing.php (hunk 1 above)
Comment (jekuaitk) [reply src/Entity/Hearing.php]: Smart, fixed in 3d4a02477b062529bd25cb3e4c1e4fbc8041c8c9.

File: src/Entity/Hearing.php (hunk 2 above)
Comment (jekuaitk) [reply src/Entity/Hearing.php]: Fixed in 3d4a02477b062529bd25cb3e4c1e4fbc8041c8c9.

File: src/Entity/HearingPost.php (hunk 3 above)
Comment (jekuaitk) [reply src/Entity/HearingPost.php]: Fixed in 3d4a02477b062529bd25cb3e4c1e4fbc8041c8c9.

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**